### ISSUE TITLE
Confusing help message for arrays

### ISSUE THREAD

**Issue by maclockard** [yargs/yargs#1349]
When arrays have a default value, in the help menu its prints as follows:

```
[array] [default: ["default-val"]]
```

This does a good job communicating that the default value is an array with a single value in it. However, if a user sees this, they assume this is how they should be structuring their arguments and try to run this literally:

```
my-cli --arg ["non-default-val"]
```

which fails.

Having these default values actually reflect usage would prevent this.

**Comment by bcoe** [comment]
@maclockard if we were to use:

`[array] [default: "default-val"]`  I would assume that the value is not an array, but a single string.

Curious if you had a recommendation for what might be less confusing.

**Comment by maclockard** [comment]
FWIW the exact issue I ran into while maintaining an internal tool was people copying and pasting `["default-val"]`, trying to modify it and complaining when it didn't work.

If only changing the help message, I would prefer `[array] [default: "default-val"]` since it passes the copy and paste test and still calls out the fact yargs is expecting an array. However, I do agree there is some potential for confusion in the other direction.

I think the best overall solution, but also the most work, would be to make it optional to use square brackets for arrays. So both `"arg1", "arg2"` and `["arg1", "arg2"]` would be valid. Not sure if this would be considered standard for cli tools, but removes any risk of folks being confused of how to properly format arrays for yargs.

**Comment by bcoe** [comment]
@maclockard I've come to hate the default behavior of arrays, where it will consume everything following `--arr a b c ...`, as this creates ambiguities in the parser ... how do you use positionals in conjunctions with arrays?

I prefer to define arrays as follows `my-app.js --arr 1 --arr 2 --arr 3`, I like that this is more explicit. It's difficult to document this advice in a way that can be copy pasted from the help output however.

**Comment by maclockard** [comment]
Yeah, the explicitness is nice. Note sure how to solve the problem of making it copy and pastable though. If you can't think of anything low lift to solve it, I'm fine with this being closed.
